openwisp_monitoring/check/tests/test_models.py

```diff
@@ -142,6 +142,24 @@ def test_config_modified_device_problem(self):
         self.assertEqual(dm.status, 'problem')
         self.assertEqual(Notification.objects.count(), 1)
 
+    @patch('openwisp_monitoring.check.settings.AUTO_PING', False)
+    def test_config_error(self):
+        """
+        Test that ConfigApplied checks are skipped when device config status is errored
+        """
+        self._create_admin()
+        self.assertEqual(Check.objects.count(), 0)
+        self._create_config(status='error', organization=self._create_org())
+        self.assertEqual(Check.objects.count(), 2)
+        self.assertEqual(Metric.objects.count(), 0)
+        self.assertEqual(AlertSettings.objects.count(), 0)
+        check = Check.objects.filter(check=self._CONFIG_APPLIED).first()
+        with freeze_time(now() - timedelta(minutes=10)):
+            check.perform_check()
+        self.assertEqual(Metric.objects.count(), 0)
+        self.assertEqual(AlertSettings.objects.count(), 0)
+        self.assertIsNone(check.perform_check())
```

Comment:
This test passes also without the fix above. Try it.
I just tried it and it's not failing.

You must always ensure the test fails without your patch.
The best thing is to add the test before the fix, alternatively comment out the fix temporarily.